onnxscript/testing/benchmark/benchmark_helpers.py

```diff
@@ -0,0 +1,682 @@
+        import onnxruntime
+        import torch
+        from onnxruntime.capi import _pybind_state as ORTC  # noqa: N812
+
+        self.sess = sess
+        self.input_names = [i.name for i in sess.get_inputs()]
+        self.output_names = [i.name for i in sess.get_outputs()]
+        self.bind = onnxruntime.SessionIOBinding(sess._sess)
+        self.OrtValue = ORTC.OrtValue
+        self.ORTC = ORTC
+        self.torch = torch
+        self.run_options = onnxruntime.RunOptions()
+
+        self.TORCH_DTYPE_TO_NUMPY_DTYPE = {
+            torch.float16: np.float16,
+            torch.float32: np.float32,
+            torch.float64: np.float64,
+            torch.uint8: np.uint8,
+            torch.int8: np.int8,
+            torch.int16: np.int16,
+            torch.int32: np.int32,
+            torch.int64: np.int64,
+            torch.bool: np.bool_,
+        }
+
+        DEVICES = {
+            -1: ORTC.OrtDevice(ORTC.OrtDevice.cpu(), ORTC.OrtDevice.default_memory(), 0)
+        }
+
+        if torch.cuda.is_available():
+            for i in range(torch.cuda.device_count()):
+                DEVICES[i] = ORTC.OrtDevice(
+                    ORTC.OrtDevice.cuda(), ORTC.OrtDevice.default_memory(), i
+                )
+
+        self.DEVICES = DEVICES
+
+    def _get_ortvalues_from_torch_tensors(
+        self,
+        tensors: tuple["torch.Tensor", ...],  # noqa: F821
+        n_outputs: int,
+        log_set: list[Any] | None = None,
+    ) -> tuple[tuple["torch.Tensor", ...], tuple["OrtDevice", ...], Any]:  # noqa: F821
+
+        ortvalues = self.ORTC.OrtValueVector()
+        ortvalues.reserve(len(tensors))
+        dtypes = []
+        shapes = []
+        data_ptrs = []
+        devices = []
+
+        max_device = -1
+        assert isinstance(max_device, int), f"unexpected type for device={max_device!r}"
+        assert tensors is not None, "tensors cannot be None"
+        new_tensors = []
+        for tensor in tensors:
+            assert isinstance(tensor, self.torch.Tensor), f"Unexpected type {type(tensor)}"
+            dtypes.append(self.TORCH_DTYPE_TO_NUMPY_DTYPE[tensor.dtype])
+            shapes.append(tensor.size())
+            data_ptrs.append(tensor.data_ptr())
+            d = tensor.get_device()
+            devices.append(self.DEVICES[d])
+            new_tensors.append(tensor)
+            max_device = max(max_device, tensor.get_device())
+
+        ortvalues.push_back_batch(new_tensors, data_ptrs, dtypes, shapes, devices)
+        output_devices = []
+        for _ in range(n_outputs):
+            dev = self.DEVICES[max_device]
+            output_devices.append(dev)
+
+        return ortvalues, output_devices
+
+    def _ortvalues_to_torch_tensor(
+        self, ortvalues: "onnxruntime.OrtValueVector"  # noqa: F821
+    ) -> tuple["torch.Tensor", ...]:  # noqa: F821
+        if len(ortvalues) == 0:
+            return tuple()
+
+        from torch._C import _from_dlpack
```

Comment:
`torch.from_dlpack` is available.

Reply:
from_dlpack is calling _from_dlpack but it is going through a test I don't need. I use this one to be faster.